### PR TITLE
Fixed updateCell to work correctly with child rows

### PR DESCRIPTION
**Pull request by andysleigh**
The updateCell method currently fails on tables with child rows. This is because when it gets the index of the row it is searching all rows (including child rows). The resulting index is therefore not the correct index to be using against the cache. Also when the update row is inserted into the cache it no longer has the child row linked to it.

The fix therefore has two parts. The first is to filter the list of rows to exclude child rows when finding the row index. The second part is to ensure the child row is linked to the updated row before inserting it to the cache.

### CODE REVIEW

**Comment by Mottie** [comment]
Hi @andysleigh!

Thanks for catching this problem and providing a fix!

File: js/jquery.tablesorter.js [hunk 1]
```diff
@@ -1328,6 +1328,11 @@
 				cache[ c.columns ].raw[ icell ] = tmp;
 				tmp = ts.getParsedText( c, cell, icell, tmp );
 				cache[ icell ] = tmp; // parsed
+				var tmpRow = cache[c.columns].$row;
+				if (tmpRow.length == 2) {
```
Comment (Mottie) [inline js/jquery.tablesorter.js]: Please change this to `if (tmpRow.length > 1) {` as it is possible to have more than one child row.

**Comment by Mottie** [comment]
Oh and please don't include the `dist` folder files... it take longer to scroll through the diff

File: js/jquery.tablesorter.js [hunk 2]
```diff
@@ -1328,6 +1328,11 @@
...
+				    // reapply child row
+				    $row = $row.add(tmpRow[1]);
```
Comment (Mottie) [inline js/jquery.tablesorter.js]: And as above, more than one child row is possible - `$row = $row.add(tmpRow);`

That "should" work, but I haven't tested it... you should be able to add all the rows back; jQuery will prevent duplication.

**Comment by Mottie** [comment]
Oh, I was thinking there was more to be done for this PR... so essentially only two changes were made:

* Add `.not( '.' + c.cssChildRow )` to find the correct row index.
* Remove the line with `cache[ c.columns ].$row = $row;`.

Is that correct?

**Comment by andysleigh** [comment]
Yes this seems to be all it needed. The first change sorts out the indexing so we get the correct row from the cache. The second part, following on from my original change I realised there's no need to update cache[...].$row at all in this method so I removed it. Sorting now works correctly after calling updateCell and the child row remains linked to the parent row.

**Comment by Mottie** [comment]
Ok great, thanks again so much for your help!

**Pull request merged**